### ISSUE TITLE
Mounting multiple encrypted file systems fails non-deterministically

### ISSUE THREAD

**Issue by prp** [lsds/sgx-lkl#531]
When running the encrypted container sample, a kernel stack trace is generated:
```
t-pepiet@msrc-cc06:~/devel/sgx-lkl-oe.git/apps/containers/encrypted$ make run-hw-verity
SGXLKL_HD_KEY=sgxlkl-alpine-crypt-verity.img.key SGXLKL_HD_VERITY=28eb36c2bbf56c9013cab79fca3dbcbe5561715c6d4526888c39e24f4917aa1f SGXLKL_HD_VERITY_OFFSET=140509184 SGXLKL_HD
_RO=1 SGXLKL_VERBOSE=1 SGXLKL_KERNEL_VERBOSE=1 ../../../build/sgx-lkl-run-oe --hw-debug sgxlkl-alpine-crypt-verity.img /usr/bin/python3 -c 'print("Hello SGX World from Encryp
ted Confidential Container!")'
[   SGX-LKL  ] SGX-LKL (OE) Git Version c3bc7ba-dirty (Debug build) [HARDWARE DEBUG]
[   SGX-LKL  ] nproc=8 SGXLKL_ETHREADS=8 SGXLKL_STHREADS=8 SGXLKL_CMDLINE=''
[   SGX-LKL  ] HW TLS support: conf->fsgsbase=1
[   SGX-LKL  ] No tap device specified, networking will not be available.
[   SGX-LKL  ] get_signed_libsgxlkl_path... result=/home/t-pepiet/devel/sgx-lkl-oe.git/build_musl/./libsgxlkl.so.signed
[   SGX-LKL  ] oe_create_enclave... result=0 (OE_OK) [   SGX-LKL  ] sgxlkl_enclave_init(ethread_id=0)
[   SGX-LKL  ] sgxlkl_ethread_init(ethread_id=2)
[   SGX-LKL  ] sgxlkl_ethread_init(ethread_id=3)
[   SGX-LKL  ] sgxlkl_ethread_init(ethread_id=4)
[   SGX-LKL  ] sgxlkl_ethread_init(ethread_id=5)
[   SGX-LKL  ] sgxlkl_ethread_init(ethread_id=6)
[   SGX-LKL  ] sgxlkl_ethread_init(ethread_id=7)
[   SGX-LKL  ] enclave base=0x7f8d80000000 size=1.042 GB
[   SGX-LKL  ] enclave heap base=0x7f8d80b18000 size=1024.00 M end=0x7f8dc0b18000
[   SGX-LKL  ] Registering OE exception handler...
[   SGX-LKL  ] kernel command line: ''
[   SGX-LKL  ] lkl_start_kernel called.
[   SGX-LKL  ] sgxlkl_ethread_init(ethread_id=1)
[    0.000000] Linux version 4.17.0+ (t-pepiet@msrc-cc06) (gcc version 7.4.0 (Ubuntu 7.4.0-1ubuntu1~18.04.1)) #180 Fri Jan 31 21:15:40 UTC 2020
[    0.000000] bootmem address range: 0x7f8dbe895000 - 0x7f8dc0894000
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 8079
[    0.000000] Kernel command line:
[    0.000000] Dentry cache hash table entries: 4096 (order: 3, 32768 bytes)
[    0.000000] Inode-cache hash table entries: 2048 (order: 2, 16384 bytes)
[    0.000000] Memory available: 32168k/32764k RAM
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 4096
[    0.000000] lkl: irqs initialized
[    0.000000] clocksource: lkl: mask: 0xffffffffffffffff max_cycles: 0x1cd42e4dffb, max_idle_ns: 881590591483 ns
[    0.000000] lkl: time and timers initialized (irq1)
[    0.000000] pid_max: default: 4096 minimum: 301
[    0.000000] Mount-cache hash table entries: 512 (order: 0, 4096 bytes)
[    0.000000] Mountpoint-cache hash table entries: 512 (order: 0, 4096 bytes)
[    0.000000] devtmpfs: initialized
[    0.004000] console [lkl_console0] enabled
[    0.004000] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.004000] xor: automatically using best checksumming function   8regs
[    0.008000] random: get_random_u32 called from 0x00007f8d80201889 with crng_init=0
[    0.008000] NET: Registered protocol family 16
[    0.008000] clocksource: Switched to clocksource lkl
[    0.008000] NET: Registered protocol family 2
[    0.008000] tcp_listen_portaddr_hash hash table entries: 256 (order: 0, 4096 bytes)
[    0.008000] TCP established hash table entries: 512 (order: 0, 4096 bytes)
[    0.008000] TCP bind hash table entries: 512 (order: 0, 4096 bytes)
[    0.008000] TCP: Hash tables configured (established 512 bind 512)
[    0.008000] UDP hash table entries: 128 (order: 0, 4096 bytes)
[    0.008000] UDP-Lite hash table entries: 128 (order: 0, 4096 bytes)
[    0.008000] NET: Registered protocol family 1
[    0.008000] workingset: timestamp_bits=62 max_order=13 bucket_order=0
[    0.012000] jitterentropy: Initialization failed with host not compliant with requirements: 2
[    0.016000] NET: Registered protocol family 38
[    0.016000] io scheduler noop registered
[    0.016000] io scheduler deadline registered
[    0.016000] io scheduler cfq registered (default)
[    0.016000] io scheduler mq-deadline registered
[    0.016000] io scheduler kyber registered
[    0.020000] loop: module loaded
[    0.020000] device-mapper: ioctl: 4.39.0-ioctl (2018-04-03) initialised: dm-devel@redhat.com
[    0.020000] wireguard: WireGuard 0.0.20190227 loaded. See www.wireguard.com for information.
[    0.020000] wireguard: Copyright (C) 2015-2019 Jason A. Donenfeld <Jason@zx2c4.com>. All Rights Reserved.
[    0.020000] NET: Registered protocol family 10
[    0.020000] Segment Routing with IPv6
[    0.020000] sit: IPv6, IPv4 and MPLS over IPv4 tunneling driver
[    0.020000] Warning: unable to open an initial console.
[    0.020000] This architecture does not have kernel memory protection.
[   SGX-LKL  ] lkl_start_kernel finished.
[   SGX-LKL  ] Adding entropy to entropy pool
[    0.024000] random: crng init done
[   SGX-LKL  ] wg0 has public key dQUGlh78LesvT+4IcsFgTnSZmubO/9mZceibBogeWEs=
[   SGX-LKL  ] aas_release_resources: deallocate all resources
[    0.024000] virtio-mmio virtio-mmio.0.auto: Failed to enable 64-bit or 32-bit DMA.  Trying to continue, but this might not work.
[    0.024000] ------------[ cut here ]------------
[    0.024000] WARNING: CPU: 0 PID: 44 at ./include/linux/dma-mapping.h:516 0x00007f8d8020ae87
[    0.024000] Call Trace:
[    0.024000] 000000005c338e9d:  [<7f8d8008b08b>] 0x00007f8d8008b08b
[    0.024000] 000000007817c70f:  [<7f8d80208e3e>] 0x00007f8d80208e3e
[    0.024000] 0000000091818e04:  [<7f8d8008b215>] 0x00007f8d8008b215
[    0.024000] 00000000651e8ecf:  [<7f8d8020ae87>] 0x00007f8d8020ae87
[    0.024000] 000000001faea9a3:  [<7f8d8020c4ef>] 0x00007f8d8020c4ef
[    0.024000] 000000008fee043c:  [<7f8d80087146>] 0x00007f8d80087146
[    0.024000] 00000000e64f6f04:  [<7f8d8020e5e8>] 0x00007f8d8020e5e8
[    0.024000] 000000004b83b1dd:  [<7f8d8020d300>] 0x00007f8d8020d300
[    0.024000] 000000007279d537:  [<7f8d802320c0>] 0x00007f8d802320c0
[    0.024000] 00000000b7d13cbe:  [<7f8d802320c0>] 0x00007f8d802320c0
[    0.024000] 0000000059f4ae3b:  [<7f8d80087146>] 0x00007f8d80087146
[    0.024000] 00000000b94408f2:  [<7f8d80232a38>] 0x00007f8d80232a38
[    0.024000] 0000000007c687a8:  [<7f8d802329eb>] 0x00007f8d802329eb
[    0.024000] 00000000d560effb:  [<7f8d8020a6f5>] 0x00007f8d8020a6f5
[    0.024000] 00000000807cd767:  [<7f8d80226f52>] 0x00007f8d80226f52
[    0.024000] 00000000f2d27538:  [<7f8d80227100>] 0x00007f8d80227100
[    0.024000] 000000000c034ee7:  [<7f8d80227183>] 0x00007f8d80227183
[    0.024000] 000000008afc8fd7:  [<7f8d80225653>] 0x00007f8d80225653
[    0.024000] 000000005a7446d6:  [<7f8d80226c0e>] 0x00007f8d80226c0e
[    0.024000] 00000000b32bb004:  [<7f8d802271be>] 0x00007f8d802271be
[    0.024000] 00000000eddf3d99:  [<7f8d80225848>] 0x00007f8d80225848
[    0.024000] 000000008c9a8155:  [<7f8d802227d4>] 0x00007f8d802227d4
[    0.024000] 000000004fe89083:  [<7f8d8020a8c4>] 0x00007f8d8020a8c4
[    0.024000] 00000000d6b6585b:  [<7f8d8020df19>] 0x00007f8d8020df19
[    0.024000] 0000000043a12056:  [<7f8d802282d4>] 0x00007f8d802282d4
[    0.024000] 00000000ff486045:  [<7f8d80226f52>] 0x00007f8d80226f52
[    0.024000] 0000000094a73ea9:  [<7f8d80227100>] 0x00007f8d80227100
[    0.024000] 00000000eed236cf:  [<7f8d80227183>] 0x00007f8d80227183
[    0.024000] 00000000bb32c0a3:  [<7f8d80225653>] 0x00007f8d80225653
[    0.024000] 000000009ddd4981:  [<7f8d80226c0e>] 0x00007f8d80226c0e
[    0.024000] 00000000b916625c:  [<7f8d802271be>] 0x00007f8d802271be
[    0.024000] 00000000fffb3162:  [<7f8d80225848>] 0x00007f8d80225848
[    0.024000] 000000002c396189:  [<7f8d802227d4>] 0x00007f8d802227d4
[    0.024000] 000000005a779fd2:  [<7f8d8008e860>] 0x00007f8d8008e860
[    0.024000] 0000000024848120:  [<7f8d802289c8>] 0x00007f8d802289c8
[    0.024000] 00000000c5723b42:  [<7f8d80087669>] 0x00007f8d80087669
[    0.024000] 00000000c57a9db1:  [<7f8d800877a3>] 0x00007f8d800877a3
[    0.024000]
[    0.024000] ---[ end trace 7881f9b20327668f ]---
[    0.024000] virtio_blk virtio0: [vda] 300647 512-byte logical blocks (154 MB/147 MiB)
[    0.096000] software IO TLB [mem 0x7f8e69798000-0x7f8e6d798000] (64MB) mapped at [000000000ced91ae-00000000f05a35b6]
[   10.799988] EXT4-fs (dm-1): mounted filesystem with ordered data mode. Opts:
[   SGX-LKL  ] Set working directory: /
[   SGX-LKL  ] Calling app main: /usr/bin/python3
Hello SGX World from Encrypted Confidential Container!
[   11.435988] reboot: Restarting system
[   SGX-LKL  ] [   SGX-LKL  ] sgxlkl_ethread_init done. ethread_id=1 result=0 (OE_OK)
[   SGX-LKL  ] [   SGX-LKL  ] sgxlkl_ethread_init done. ethread_id=7 result=0 (OE_OK)
vio_enclave_process_host_event: [ dev_id = 0 ] exit
[   SGX-LKL  ] [   SGX-LKL  ] [   SGX-LKL  ] sgxlkl_ethread_init done. ethread_id=3 result=0 (OE_OK)
[   SGX-LKL  ] sgxlkl_ethread_init done. ethread_id=2 result=0 (OE_OK)
vio_enclave_process_host_event: [ dev_id = 2 ] exit
sgxlkl_ethread_init done. ethread_id=5 result=0 (OE_OK)
vio_enclave_process_host_event: [ dev_id = 1 ] exit
[   SGX-LKL  ] sgxlkl_ethread_init done. ethread_id=6 result=0 (OE_OK)
[   SGX-LKL  ] sgxlkl_ethread_init done. ethread_id=4 result=0 (OE_OK)
[   SGX-LKL  ] sgxlkl_enclave_init done. ethread_id=0 result=0 (OE_OK)
[   SGX-LKL  ] oe_terminate_enclave... done
[   SGX-LKL  ] SGX-LKL-OE exit
t-pepiet@msrc-cc06:~/devel/sgx-lkl-oe.git/apps/containers/encrypted$
```

The same behaviour occurs for `make run-hw-integrity`.

This should be fixed by https://github.com/lsds/sgx-lkl/issues/98.

**Comment by prp** [comment]
I've reactivated the tests in #728. If this passes, we can close this issue.

**Comment by prp** [comment]
Fixed by #697 (see #728).
